zebra-chain/src/sapling/keys.rs

```diff
@@ -505,10 +505,15 @@ impl From<AuthorizingKey> for [u8; 32] {
     }
 }
 
-impl From<SpendAuthorizingKey> for AuthorizingKey {
-    fn from(ask: SpendAuthorizingKey) -> Self {
-        let sk = redjubjub::SigningKey::<SpendAuth>::try_from(<[u8; 32]>::from(ask)).unwrap();
```

Comment:
@conradoplg do you think this is the same as https://github.com/ZcashFoundation/zebra/pull/4609/files#r899478519 ?

Reply:
Yep, it's the same

Reply:
So i think this is what we will end up doing, let me know how do you see it:

- Leave orchard and sapling files as they are but add `expect` instead of unwrap in those 2 lines that we discussed. This will also leave unchanged the address and the test files.
- Allow the lint in the above files.
- We can keep the lint active in protocol messages.
- We can keep the exception in u256.

So basically we will be introducing a lint but allowing almost all the code that the lint forbid. Will this worth ?

Reply:
I think the lint is still useful, because it will make us carefully review any panicking `From` impls that we add to Zebra.

(Sometimes the reviews will show that they are ok, and we'll add a comment and allow the lint. That is also a good outcome.)

Reply:
Sounds good.